CMakeLists.txt

```diff
@@ -150,19 +157,38 @@ add_dependencies(examples examples_by_runtime_api_module modified_cuda_samples)
 
 install(
 	TARGETS cuda-api-wrappers
-	EXPORT cuda-api-wrappers
+	EXPORT cuda-api-wrappersExport
 	RUNTIME DESTINATION bin
 	ARCHIVE DESTINATION lib
 	LIBRARY DESTINATION lib
 	INCLUDES DESTINATION include
-	CONFIGURATIONS Release RelWithDebugInfo
 )
 
 install(
-#	EXPORT cuda-api-wrappers
 	DIRECTORY src/cuda
 	DESTINATION include
 	FILES_MATCHING REGEX "\\.(h|hpp|cuh)"
 )
 
-export(EXPORT cuda-api-wrappers FILE cuda-api-wrappers.cmake)
+install(
+	EXPORT ${PROJECT_NAME}Export
+	DESTINATION "share/cmake/${PROJECT_NAME}"
+	NAMESPACE "${PROJECT_NAME}::"
+	# In this CMake config file no dependencies are considered. But since we
+	# do not use any `find_package` call here this approach is sufficient.
+	FILE ${PROJECT_NAME}Config.cmake
+)
+
+if(EXPORT_BUILD_DIR AND NOT CMAKE_EXPORT_NO_PACKAGE_REGISTRY)
+	message("-- Exporting ${PROJECT_NAME} build directory to local CMake package registry.")
+
+	export(
+		EXPORT ${PROJECT_NAME}Export
+		NAMESPACE "${PROJECT_NAME}::"
+		# In this CMake config file no dependencies are considered. But since we
+		# do not use any `find_package` call here this approach is sufficient.
+		FILE "${PROJECT_BINARY_DIR}/${PROJECT_NAME}Config.cmake"
```

Comment:
Again, shouldn't this be either `cuda-api-wrappers_config` or `CudaApiWrappersConfig.cmake`

Reply:
How do you want the user to add the dependency to this project. I think the most intuitive way would be (this is how it is working right now):

```
find_package(cuda-api-wrappers)
...
target_add_libraries(
  foo
  cuda-api-wrappers::cuda-api-wrappers
)
```

If you name this file `CudaApiWrappersConfig.cmake` it would be

```
find_package(CudaApiWrappers)
...
target_add_libraries(
  foo
  cuda-api-wrappers::cuda-api-wrappers
)
```

and if you name this file `cuda-api-wrappers_config` it is useless, because CMake expects `-config` or `Config` suffix.